File: java/file-crypto-auto-encrypt/src/main/java/com/ionic/samples/FileCryptoAutoEncrypt.java
/*
 * (c) 2017 Ionic Security Inc.
 * By using this code, I agree to the Terms & Conditions (https://www.ionic.com/terms-of-use/)
 * and the Privacy Policy (https://www.ionic.com/privacy-notice/).
 */

import com.ionicsecurity.sdk.Agent;
import com.ionicsecurity.sdk.AgentSdk;
import com.ionicsecurity.sdk.AutoFileCipher;
import com.ionicsecurity.sdk.DeviceProfilePersistorPlainText;
import com.ionicsecurity.sdk.FileCryptoEncryptAttributes;
import com.ionicsecurity.sdk.KeyAttributesMap;
import java.util.List;
import java.util.LinkedList;

public class FileCryptoAutoEncrypt
{
  public static void main(String[] args)
  {
    if (args.length != 2) {
      System.out.println("Usage args: [pathToPlaintext] [pathToTargetCiphertext]");
      return;
    }
    String pathToPlainFile = args[0];
    String pathToEncryptedFile = args[1]; 
    
    AgentSdk.initialize(null);  
    Agent agent = new Agent();
    if (System.getProperty("os.name").startsWith("Linux")) {
      DeviceProfilePersistorPlainText ptPersistor = new DeviceProfilePersistorPlainText();
      String sProfilePath = System.getProperty("user.home") + "/.ionicsecurity/profiles.pt";
      ptPersistor.setFilePath(sProfilePath);
      agent.initialize(ptPersistor);
    } else {
      agent.initialize();
    }
    
    AutoFileCipher fileCipher = new AutoFileCipher(agent);
    
    FileCryptoEncryptAttributes encryptAttributes = new FileCryptoEncryptAttributes();

    KeyAttributesMap attrMap = new KeyAttributesMap();
    List values = new ArrayList<String>();
    values.add("restricted"); //make sure you have access to keys marked as restricted before you do add this key attribute
    attrMap.set("classification", values);
    encryptAttributes.setKeyAttributes(attrMap);
    
    fileCipher.encrypt(pathToPlainFile, pathToEncryptedFile);
  }
}

File: python/file-crypto-auto-encrypt/filecryptoautoencrypt.py
#!/usr/bin/env python

# (c) 2017 Ionic Security Inc.
# By using this code, I agree to the Terms & Conditions (https://www.ionic.com/terms-of-use/)
# and the Privacy Policy (https://www.ionic.com/privacy-notice/).

import ionicsdk

import sys
import platform
import os.path

#NOTE: You MUST have the input file present before running this:
INPUT_FILE_PATH = "example_file.docx"
OUTPUT_FILE_PATH = "example_file_encrypted.docx"

def get_agent():
  # Load a plaintext Device Profile, or SEP, from disk if on Linux (e.g., no default persistor).
  persistor = None
  if platform.system == 'Linux':
    SEP_FILE_PATH = os.path.expanduser("~/.ionicsecurity/profiles.pt")
    persistor = ionicsdk.DeviceProfilePersistorPlaintextFile(SEP_FILE_PATH)
    print("A plaintext SEP will be loaded from {0}".format(SEP_FILE_PATH))

  # Initialize the Ionic Agent (must be done before most Ionic operations).
  agent = ionicsdk.Agent(None, persistor)

  # Check if there are profiles and that there is one set as active.
  if not agent.hasanyprofiles() or not agent.hasactiveprofile():
    if not agent.hasanyprofiles():
      print("There are no device profiles on this device.")
    if not agent.hasactiveprofile():
      print("There is not an active device profile selected on this device.")
    print("Register (and select an active profile) this device before continuing.")
    sys.exit(-1)

  return agent

if __name__ == "__main__":
  agent = get_agent()
  if agent is None:
    sys.exit(-1)

  # Create a classification (or any other key attributes to set)
  key_attrs = {"classification": "Restricted"}

  # Encrypt the file using the classification
  cipher = ionicsdk.FileCipherAuto(agent)
  try:
    cipher.encrypt(INPUT_FILE_PATH, OUTPUT_FILE_PATH, attributes=key_attrs)
    #NOTE: Due to the format of CSV files, FileCipherAuto can't automatically detect if a file should be processed
    # as a CSV file vs a generic file format. To handle this, use the FileCrypto.getinfo capability to determine it isn't
    # another type of file, and then check the input filename's extension to match "csv" and use a FileCipherCsv cipher.
  except ionicsdk.exceptions.IonicException as e:
    print("Error encrypting: {0}".format(e.message))
    sys.exit(-2)

  print("Encrypted file {0} to {1}.".format(INPUT_FILE_PATH, OUTPUT_FILE_PATH))
